python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -189,6 +190,24 @@ def from_coercible_or_definition(
         else:
             return AssetKey.from_coercible(arg)
 
+    @staticmethod
+    def from_coercible_to_asset_dep(arg: "CoercibleToAssetDep") -> "AssetKey":
+        from dagster._core.definitions.asset_dep import AssetDep
+        from dagster._core.definitions.asset_spec import AssetSpec
+        from dagster._core.definitions.assets import AssetsDefinition
+        from dagster._core.definitions.source_asset import SourceAsset
+
+        if isinstance(arg, AssetsDefinition):
+            return arg.key
```

Comment:
hmmm do we need explict error here if this is multi-key?

Reply:
yes, good catch - ty